core/templates/dev/head/filtersSpec.js

```diff
@@ -483,6 +484,14 @@ describe('Testing filters', function() {
       'Text Input 3');
     }));
 
+  it('should remove extra new lines', inject(function($filter) {
+    var filter = $filter('removeExtraLines');
+
+    expect(filter('<p><br></p>')).toEqual('');
```

Comment:
I think we need more tests, including more realistic content. For example, a test to ensure that `<p><br></p>` is only filtered out if it is at the end of a string, and not anywhere else. A test for when there are multiple empty lines at the end of a string instead of just one, etc. etc.